pkg/stockings/fingopack.go

```diff
@@ -0,0 +1,159 @@
+	// finance.quote has a HUGE variety of info in it that we can pull
+	symbol := quote.Symbol
+	name := quote.ShortName
+	currentPrice := decimal.NewFromFloat(quote.RegularMarketPrice)
+	priceChange := float32(quote.RegularMarketChange)
+	percentChange := float32(quote.RegularMarketChangePercent)
+
+	return &Stock{symbol, name, currentPrice, priceChange, percentChange}, nil
+
+}
+
+// GetHistoricalPrice is hacky because Fingo only has a function that returns quotes in a time span
+// This function takes in the desired date as the end date, then sets the start date as one month prior
+func (piq FingoPack) GetHistoricalPrice(ticker string, date time.Time) (*HistoricalStock, error) {
+
+	quoteAtDate := new(finance.ChartBar)
+
+	// converting the time.Time struct into usuable integers to be fed into Fingo
+	year := date.Year()
+	month := int(date.Month())
+	day := date.Day()
+
+	// converting into our accepted Fingo form
+	endDate := datetime.Datetime{
+		Year:  year,
+		Month: month,
+		Day:   day,
+	}
+
+	// this brings our month down by one
+	if month == 1 {
+		month = 12
+		year--
+	}
+
+	// now we can set our start date which is just one month prior - perhaps this part should be put in a seperate function
+	startDate := datetime.Datetime{
+		Year:  year,
+		Month: month,
+		Day:   day,
+	}
+
+	// now we are using the Fingo steps to get historical stock info in a range
+	// set our paratemters which are self explanatory
+	params := &chart.Params{
+		Symbol:   ticker,
+		Interval: datetime.OneDay,
+		Start:    &startDate,
+		End:      &endDate,
+	}
+
+	// a little bit of blackboxing here, but this iter contains all the information we need
+	iter := chart.Get(params)
+
+	// iter.Next() eventually equals false when you've finished
+	for iter.Next() {
+		// we are only interested in saving the last value in our iterator because that is reflective of the correct price requested by user
+		// note if the user requests a weekend, then we need to return the previous adj close price
+		quoteAtDate = iter.Bar()
+	}
+
+	// some more blackboxing...somehow this just returns errors :)
+	err := iter.Err()
+	if err != nil {
+		return nil, err
+	}
+
+	// at this point quoteAtDate holds the finance.ChartBar struct of information we need
+	// just need to convert it into our universal HistoricalStock struct
+	return &(HistoricalStock{time.Unix(int64(quoteAtDate.Timestamp), 0), quoteAtDate.AdjClose}), nil
+
+}
+
+// GetHistoricalRange is nearly identical to GetHistoricalPrice, but there are some nuances - might not be able to create a seperate helper function
+func (piq FingoPack) GetHistoricalRange(ticker string, start time.Time, end time.Time) (*HistoricalStocks, error) {
+
+	// need to create a piqHistoricalStocks type that is a list of finance.ChartBar
+	historicalRange := new(piqHistoricalStocks)
+
+	// just setting our start and end date based on what we were given
+	startDate := datetime.Datetime{
```

Comment:
same here, you can just use their `datetime.New()` function

Reply:
done